lib/gen_stage.ex

```diff
@@ -2295,12 +2312,21 @@ defmodule GenStage do
     buffer_events(events, stage)
   end
 
-  defp take_from_buffer(counter, %{buffer: buffer} = stage) do
+  defp take_from_buffer(
+         counter,
+         %{buffer: buffer, mod: mod, mod_handle_buffered: mod_handle_buffered, state: state} =
+           stage
+       ) do
     case Buffer.take_count_or_until_permanent(buffer, counter) do
       :empty ->
         {:ok, counter, stage}
 
       {:ok, buffer, new_counter, temps, perms} ->
+        if mod_handle_buffered do
```

Comment:
do you think it would make more sense if this callback is called by the Buffer module whenever the buffer size changes?

Reply:
I thought it might, but `Buffer` is really generic and it felt like this was the right level of abstraction because the existing log is at this level.